### PR TITLE
add RHUI test for Redhat

### PR DESCRIPTION
**Pull request by Nidylei**
1. add RHUI test for Redhat
2. add a sudo user when prepared image
3. minor fix

### CODE REVIEW

File: remote-scripts/SETUP-INSTALL-PACKAGES.ps1 [hunk 1]
```diff
@@ -71,6 +71,24 @@ if ($isDeployed)
...
+						$newUserAdded = $true
+						Set-Content -Value $userAddOutput -Path $LogDir\userAddOutput.txt -Force
+						LogMsg "Add new user : $newUser : SUCCESS"
+						LogMsg "Password for : $newUser : $newPassword"
+					}
+					else
+					{
+						$newUserAdded = $false
+						LogErr "Add new user : $newUser : FAILED"
+						LogErr "Output : $userAddOutput"
```
Comment (lizzha) [inline remote-scripts/SETUP-INSTALL-PACKAGES.ps1]: Adding user failed should not fail the test, the failure is most likely not image issue, but the test code issue

Comment (Nidylei) [inline remote-scripts/SETUP-INSTALL-PACKAGES.ps1]: Yes, actually the test indeed won't fail no matter the user is added failed or successful. Just generate some error log!   I have try it on each distro , the user can be added successful.

File: Azure_ICA_all.xml [hunk 2]
```diff
@@ -2292,7 +2292,7 @@
 			<testName>SETUP-INSTALL-PACKAGES</testName>
 			<testScript>SETUP-INSTALL-PACKAGES.py</testScript>
 			<testScriptPs1>SETUP-INSTALL-PACKAGES.ps1</testScriptPs1>
-			<files>.\remote-scripts\azuremodules.py,.\remote-scripts\SETUP-INSTALL-PACKAGES.py,.\remote-scripts\packages.xml</files>
+			<files>.\remote-scripts\azuremodules.py,.\remote-scripts\SETUP-INSTALL-PACKAGES.py,.\remote-scripts\AddNewUserWithSudoPremission.sh,.\remote-scripts\packages.xml</files>
```
Comment (LiliDeng) [inline Azure_ICA_all.xml]: Premission => Permission

File: remote-scripts/SETUP-INSTALL-PACKAGES.ps1 [hunk 3]
```diff
@@ -71,6 +71,24 @@ if ($isDeployed)
 					{
 						$usePython3 = $true
 					}
+					#Add a User with sudo permissions for investigating issue
+					$newUser = 'shostctest'
+					$newPassword = 'P@ssw0rd01'
```
Comment (lizzha) [inline remote-scripts/SETUP-INSTALL-PACKAGES.ps1]: Can we make it a parameter in CI site?? It's not safe to expose the username/password in public

File: remote-scripts/SETUP-INSTALL-PACKAGES.py [hunk 4]
```diff
@@ -156,6 +158,38 @@ def install_ez_setup():
         output = Run("{0} {1}".format(python_cmd, ez_setup))
         return ("Finished" in output)
 
+def InstallGcc():
```
Comment (lizzha) [inline remote-scripts/SETUP-INSTALL-PACKAGES.py]: why adding this method?

Comment (Nidylei) [inline remote-scripts/SETUP-INSTALL-PACKAGES.py]: For SLES12, gcc installation need several interactive operation, and some option can't be the default value, so add this method to fix it.

**Pull request merged**